spec.bs

```diff
@@ -1364,6 +1414,25 @@ The <dfn attribute for="AppHistoryDestination">sameDocument</dfn> getter steps a
   1. For each |traversal| of |traversals|: [=finalize with an aborted navigation error=] given |appHistory| and |traversal|.
 </div>
 
+<div algorithm>
+  To <dfn>potentially reset the focus</dfn> given an {{AppHistory}} object |appHistory| and an {{AppHistoryNavigateEvent}} |event|:
+
+  1. Let |focusChanged| be |appHistory|'s [=AppHistory/focus changed during ongoing navigation=].
+  1. Set |appHistory|'s [=AppHistory/focus changed during ongoing navigation=] to false.
+  1. If |event|'s [=AppHistoryNavigateEvent/focus reset behavior=] is "{{AppHistoryFocusReset/manual}}", then return.
+     <p class="note">If it was left as null, then we treat that was "{{AppHistoryFocusReset/after-transition}}", and continue onward.
+  1. If |focusChanged| is true, then return.
+  1. Let |document| be |appHistory|'s [=relevant global object=]'s [=associated Document=].
+  1. Let |focusTarget| be the <a spec="HTML">autofocus delegate</a> for |document|.
+  1. If |focusTarget| is null, then set |focusTarget| to |document|'s <a spec="HTML" lt="the body element">body element</a>.
+  1. If |focusTarget| is null, then set |focusTarget| to |document|'s [=document element=].
+  1. Run the <a spec="HTML">focusing steps</a> for |focusTarget|, with |document|'s <a spec="CSS2">viewport</a> as the fallback target.
```

Comment:
We should also update the sequential focus navigation starting point, I think?

Specs currently don't update it. Blink updates it whenever focusing an element, but *doesn't* (without extra code) update it when resetting focus to the body.